qiskit/aqua/operators/primitive_ops/pauli_sum_op.py

```diff
@@ -0,0 +1,405 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" PauliSumOp Class """
+
+import logging
+from numbers import Number
+from typing import Dict, List, Optional, Set, Tuple, Union, cast
+
+import numpy as np
+from scipy.sparse import spmatrix
+
+from qiskit.circuit import Instruction, ParameterExpression
+from qiskit.quantum_info import Pauli, SparsePauliOp
+
+from ... import AquaError
+from ..list_ops.summed_op import SummedOp
+from ..list_ops.tensored_op import TensoredOp
+from ..operator_base import OperatorBase
+from .primitive_op import PrimitiveOp
+
+logger = logging.getLogger(__name__)
+
+
+class PauliSumOp(PrimitiveOp):
+    """Class for Operators backend by Terra's ``SparsePauliOp`` class."""
+
+    def __init__(
+            self,
+            primitive: SparsePauliOp,
+            coeff: Union[int, float, complex, ParameterExpression] = 1.0,
+    ) -> None:
+        """
+        Args:
+            primitive: The SparsePauliOp which defines the behavior of the underlying function.
+            coeff: A coefficient multiplying the primitive.
+
+        Raises:
+            TypeError: invalid parameters.
+        """
+        if not isinstance(primitive, SparsePauliOp):
+            raise TypeError(
+                "PauliSumOp can only be instantiated with SparsePauliOp, "
+                f"not {type(primitive)}"
+            )
+
+        super().__init__(primitive, coeff=coeff)
+
+    def primitive_strings(self) -> Set[str]:
+        return {"SparsePauliOp"}
+
+    @property
+    def num_qubits(self) -> int:
+        return self.primitive.num_qubits  # type: ignore
```

Comment:
Maybe it's better to do a cast here instead of ignoring, then this will be fixed in checks that depend on this type
```suggestion
        return cast(int, self.primitive.num_qubits)
```